src/NServiceBus.Core/Pipeline/MainPipelineExecutor.cs

```diff
@@ -48,11 +54,27 @@ public async Task Invoke(MessageContext messageContext, CancellationToken cancel
 
                     ex.Data["Pipeline canceled"] = transportReceiveContext.CancellationToken.IsCancellationRequested;
 
+                    // TODO: Add an explicit tag for operation canceled
+                    ActivityDecorator.SetErrorStatus(activity, ex);
                     throw;
                 }
 
                 await receivePipelineNotification.Raise(new ReceivePipelineCompleted(message, pipelineStartedAt, DateTimeOffset.UtcNow), cancellationToken).ConfigureAwait(false);
             }
+
+            activity?.SetStatus(ActivityStatusCode.Ok); //Set acitivity state.
+        }
+
+        static Activity CreateIncomingActivity(MessageContext context)
+        {
+            //TODO Do we need to check for Activity.Current first in case the transport creates it's own span?
```

Comment:
The transport and/or the native client could create a span. This is why meta data shouldn't always be added to current activity but to the root of the activity that represents the unit-of-work.